# Style Transfer model training, converting to ml5.js/tf.js and video stlye transfer

This notebook is based on the [Google Colab notebook](https://colab.research.google.com/drive/1TZNdhoUEBoxQqY5EFloZcuyYUs9oNQ-g), provided by ml5 and adapted so it works with Paperspace Gradient notebooks.  

It excludes downloading of the 14GB+ COCO image dataset, since it is already provided by Paperspace.  

First, we prepare the environment by downloading the fast-style-transfer git repo and the vgg image, installing libraries that are not included in the Paperspace TensorFlow 1.14 Docker image, train the model, convert it into ml5 appropriate format and (optionally) use it to perform stlye transfer on a video.

NOTE: For this notebook to work, the Paperspace Docker image must be TensorFlow 1.14.


## 1. Preparing the environment


In [ ]:
# Upgrade pip
pip install --upgrade pip

In [ ]:
# Clone the fast-style-transfer git repo from github: https://github.com/lengstrom/fast-style-transfer.
!git clone https://github.com/lengstrom/fast-style-transfer

Cloning into 'fast-style-transfer'...


remote: Enumerating objects: 248, done.
remote: Counting objects:  10% (1/10)
remote: Counting objects:  20% (2/10)
remote: Counting objects:  30% (3/10)
remote: Counting objects:  40% (4/10)
remote: Counting objects:  50% (5/10)
remote: Counting objects:  60% (6/10)
remote: Counting objects:  70% (7/10)
remote: Counting objects:  80% (8/10)
remote: Counting objects:  90% (9/10)
remote: Counting objects: 100% (10/10)
remote: Counting objects: 100% (10/10), done.
remote: Compressing objects:  10% (1/10)
remote: Compressing objects:  20% (2/10)
remote: Compressing objects:  30% (3/10)
remote: Compressing objects:  40% (4/10)
remote: Compressing objects:  50% (5/10)
remote: Compressing objects:  60% (6/10)
remote: Compressing objects:  70% (7/10)
remote: Compressing objects:  80% (8/10)
remote: Compressing objects:  90% (9/10)
remote: Compressing objects: 100% (10/10)
remote: Compressing objects: 100% (10/10), done.


Receiving objects:   0% (1/248)   
Receiving objects:   1% (3/248)   
Receiving objects:   2% (5/248)   
Receiving objects:   3% (8/248)   
Receiving objects:   4% (10/248)   
Receiving objects:   5% (13/248)   
Receiving objects:   6% (15/248)   
Receiving objects:   7% (18/248)   
Receiving objects:   8% (20/248)   
Receiving objects:   9% (23/248)   
Receiving objects:  10% (25/248)   
Receiving objects:  11% (28/248)   
Receiving objects:  12% (30/248)   
Receiving objects:  13% (33/248)   
Receiving objects:  14% (35/248)   
Receiving objects:  15% (38/248)   
Receiving objects:  16% (40/248)   
Receiving objects:  17% (43/248)   
Receiving objects:  18% (45/248)   
Receiving objects:  19% (48/248)   
Receiving objects:  20% (50/248)   
Receiving objects:  21% (53/248)   
Receiving objects:  22% (55/248)   
Receiving objects:  23% (58/248)   
Receiving objects:  24% (60/248)   
Receiving objects:  25% (62/248)   
Receiving objects:  26% (65/248)   
Receiving objects:  27% (67/248)

Receiving objects:  37% (92/248), 7.37 MiB | 14.37 MiB/s   
Receiving objects:  38% (95/248), 7.37 MiB | 14.37 MiB/s   
Receiving objects:  39% (97/248), 7.37 MiB | 14.37 MiB/s   


Receiving objects:  41% (102/248), 7.37 MiB | 14.37 MiB/s   
remote: Total 248 (delta 2), reused 5 (delta 0), pack-reused 238
Receiving objects:  42% (105/248), 7.37 MiB | 14.37 MiB/s   
Receiving objects:  43% (107/248), 7.37 MiB | 14.37 MiB/s   
Receiving objects:  44% (110/248), 7.37 MiB | 14.37 MiB/s   
Receiving objects:  45% (112/248), 7.37 MiB | 14.37 MiB/s   
Receiving objects:  46% (115/248), 7.37 MiB | 14.37 MiB/s   
Receiving objects:  47% (117/248), 7.37 MiB | 14.37 MiB/s   
Receiving objects:  48% (120/248), 7.37 MiB | 14.37 MiB/s   
Receiving objects:  49% (122/248), 7.37 MiB | 14.37 MiB/s   
Receiving objects:  50% (124/248), 7.37 MiB | 14.37 MiB/s   
Receiving objects:  51% (127/248), 7.37 MiB | 14.37 MiB/s   
Receiving objects:  52% (129/248), 7.37 MiB | 14.37 MiB/s   
Receiving objects:  53% (132/248), 7.37 MiB | 14.37 MiB/s   
Receiving objects:  54% (134/248), 7.37 MiB | 14.37 MiB/s   
Receiving objects:  55% (137/248), 7.37 MiB | 14.37 MiB/s   
Receiving objects:  

### Creating directories

Create a folder called 'ckpt' inside of 'fast-style-transfer' folder

In [ ]:
%mkdir /notebooks/fast-style-transfer/ckpt

Create another folder called 'images' inside of 'fast-style-transfer' folder

In [ ]:
%mkdir /notebooks/fast-style-transfer

Inside of the 'images' folder, create a folder called 'style'

In [ ]:
%mkdir /notebooks/fast-style-transfer/images/style

### Add a style image
- Go to left sidebar, click on the folder icon to open "Files" panel
- Put a style image inside of the _fast-style-transfer/images/style_ folder

### Install additional libraries

#### Install a higher version of tensorflow for training

We need the 2.1.0 version for training and video style transfer.  
The TensorFlow 1.14 version is needed only to convert the trained model to ml5, but we can only upgrade a version in a Docker image and not downgrade - this is why we start with the 1.14 image.

In [ ]:
# Install a higher version of tensorflow for training
!pip install tensorflow-gpu==2.1.0

#### Install necessary libraries

In [ ]:
# Install imageio
!pip install imageio

#### Install libraries for video style transfer (optional)

In [ ]:
# Install libraries for video style transfer
!pip install ffmpeg
!pip install moviepy==1.0.2
!pip install scipy
# Downgrading the image-ffmpeg prevent some pipe error
!pip install imageio-ffmpeg==0.2.0

## 2. Preparing the training data

#### Check that the COCO dataset exists on Paperspace

In [ ]:
# Check for the coco dataset in the paperspace files
%ls /datasets/coco/

annotations/  coco_train2014/  coco_val2014/


#### Downloading the vgg .mat file

In [ ]:
# Create the necessary directory
%mkdir /notebooks/fast-style-transfer/data/

In [ ]:
# Go into the appropriate directory and download the necessary file
%cd /notebooks/fast-style-transfer/data/
!wget https://www.vlfeat.org/matconvnet/models/beta16/imagenet-vgg-verydeep-19.mat

/notebooks/fast-style-transfer/data


--2021-12-10 15:55:21--  https://www.vlfeat.org/matconvnet/models/beta16/imagenet-vgg-verydeep-19.mat
Resolving www.vlfeat.org (www.vlfeat.org)... 

64.90.48.57
Connecting to www.vlfeat.org (www.vlfeat.org)|64.90.48.57|:443... 

connected.


HTTP request sent, awaiting response... 

200 OK
Length: 576042600 (549M)


Saving to: ‘imagenet-vgg-verydeep-19.mat’


          imagenet-   0%[                    ]       0  --.-KB/s               


         imagenet-v   0%[                    ] 200.00K   966KB/s               


        imagenet-vg   0%[                    ] 744.00K  1.63MB/s               


       imagenet-vgg   0%[                    ]   3.07M  4.74MB/s               


      imagenet-vgg-   0%[                    ]   4.74M  5.59MB/s               


     imagenet-vgg-v   1%[                    ]   6.40M  6.10MB/s               


    imagenet-vgg-ve   1%[                    ]   8.23M  6.58MB/s               


   imagenet-vgg-ver   1%[                    ]   9.99M  6.87MB/s               


  imagenet-vgg-very   2%[                    ]  11.88M  7.18MB/s               


 imagenet-vgg-veryd   2%[                    ]  13.82M  7.45MB/s               


imagenet-vgg-veryde   2%[                    ]  15.80M  7.59MB/s               


magenet-vgg-verydee   3%[                    ]  17.74M  7.77MB/s               


agenet-vgg-verydeep   3%[                    ]  19.79M  7.97MB/s               


genet-vgg-verydeep-   3%[                    ]  21.87M  8.15MB/s               


enet-vgg-verydeep-1   4%[                    ]  23.93M  8.29MB/s               


net-vgg-verydeep-19   4%[                    ]  25.49M  8.26MB/s    eta 63s    


et-vgg-verydeep-19.   4%[                    ]  27.30M  8.29MB/s    eta 63s    


t-vgg-verydeep-19.m   5%[>                   ]  27.88M  8.42MB/s    eta 63s    


-vgg-verydeep-19.ma   5%[>                   ]  28.76M  8.63MB/s    eta 63s    


vgg-verydeep-19.mat   5%[>                   ]  29.77M  8.02MB/s    eta 63s    


gg-verydeep-19.mat    5%[>                   ]  30.88M  7.75MB/s    eta 71s    


g-verydeep-19.mat     5%[>                   ]  31.66M  7.37MB/s    eta 71s    


-verydeep-19.mat      5%[>                   ]  32.60M  7.05MB/s    eta 71s    


verydeep-19.mat       6%[>                   ]  33.27M  6.70MB/s    eta 71s    


erydeep-19.mat        6%[>                   ]  34.18M  6.51MB/s    eta 71s    


rydeep-19.mat         6%[>                   ]  35.01M  6.21MB/s    eta 78s    


ydeep-19.mat          6%[>                   ]  35.57M  5.88MB/s    eta 78s    


deep-19.mat           6%[>                   ]  36.40M  5.59MB/s    eta 78s    


eep-19.mat            6%[>                   ]  36.95M  5.22MB/s    eta 78s    


ep-19.mat             6%[>                   ]  37.59M  4.67MB/s    eta 78s    


p-19.mat              6%[>                   ]  38.34M  4.36MB/s    eta 84s    


-19.mat               7%[>                   ]  38.96M  3.94MB/s    eta 84s    


19.mat                7%[>                   ]  39.62M  3.62MB/s    eta 84s    


9.mat                 7%[>                   ]  40.32M  3.53MB/s    eta 84s    


.mat                  7%[>                   ]  41.15M  3.54MB/s    eta 84s    


mat                   7%[>                   ]  41.95M  3.55MB/s    eta 90s    


at                    7%[>                   ]  42.70M  3.51MB/s    eta 90s    


t                     7%[>                   ]  43.43M  3.45MB/s    eta 90s    


                      8%[>                   ]  44.26M  3.42MB/s    eta 90s    


                  i   8%[>                   ]  45.12M  3.53MB/s    eta 90s    


                 im   8%[>                   ]  45.87M  3.46MB/s    eta 93s    


                ima   8%[>                   ]  46.71M  3.48MB/s    eta 93s    


               imag   8%[>                   ]  47.43M  3.42MB/s    eta 93s    


              image   8%[>                   ]  48.21M  3.43MB/s    eta 93s    


             imagen   8%[>                   ]  48.98M  3.48MB/s    eta 93s    


            imagene   9%[>                   ]  49.76M  3.45MB/s    eta 96s    


           imagenet   9%[>                   ]  50.60M  3.54MB/s    eta 96s    


          imagenet-   9%[>                   ]  51.41M  3.53MB/s    eta 96s    


         imagenet-v   9%[>                   ]  52.21M  3.59MB/s    eta 96s    


        imagenet-vg   9%[>                   ]  52.99M  3.67MB/s    eta 96s    


       imagenet-vgg   9%[>                   ]  53.84M  3.75MB/s    eta 98s    


      imagenet-vgg-   9%[>                   ]  54.66M  3.74MB/s    eta 98s    


     imagenet-vgg-v  10%[=>                  ]  55.45M  3.75MB/s    eta 98s    


    imagenet-vgg-ve  10%[=>                  ]  56.23M  3.77MB/s    eta 98s    


   imagenet-vgg-ver  10%[=>                  ]  57.02M  3.78MB/s    eta 98s    


  imagenet-vgg-very  10%[=>                  ]  57.87M  3.81MB/s    eta 99s    


 imagenet-vgg-veryd  10%[=>                  ]  58.65M  3.79MB/s    eta 99s    


imagenet-vgg-veryde  10%[=>                  ]  59.41M  3.77MB/s    eta 99s    


magenet-vgg-verydee  10%[=>                  ]  60.24M  3.77MB/s    eta 99s    


agenet-vgg-verydeep  11%[=>                  ]  61.09M  3.85MB/s    eta 99s    


genet-vgg-verydeep-  11%[=>                  ]  61.88M  3.82MB/s    eta 1m 40s 


enet-vgg-verydeep-1  11%[=>                  ]  62.65M  3.84MB/s    eta 1m 40s 


net-vgg-verydeep-19  11%[=>                  ]  63.41M  3.84MB/s    eta 1m 40s 


et-vgg-verydeep-19.  11%[=>                  ]  64.20M  3.79MB/s    eta 1m 40s 


t-vgg-verydeep-19.m  11%[=>                  ]  64.76M  3.72MB/s    eta 1m 40s 


-vgg-verydeep-19.ma  11%[=>                  ]  65.73M  3.75MB/s    eta 1m 41s 


vgg-verydeep-19.mat  12%[=>                  ]  66.37M  3.71MB/s    eta 1m 41s 


gg-verydeep-19.mat   12%[=>                  ]  67.09M  3.67MB/s    eta 1m 41s 


g-verydeep-19.mat    12%[=>                  ]  67.90M  3.71MB/s    eta 1m 41s 


-verydeep-19.mat     12%[=>                  ]  68.65M  3.63MB/s    eta 1m 41s 


verydeep-19.mat      12%[=>                  ]  69.40M  3.57MB/s    eta 1m 43s 


erydeep-19.mat       12%[=>                  ]  70.16M  3.60MB/s    eta 1m 43s 


rydeep-19.mat        12%[=>                  ]  70.80M  3.54MB/s    eta 1m 43s 


ydeep-19.mat         13%[=>                  ]  71.68M  3.57MB/s    eta 1m 43s 


deep-19.mat          13%[=>                  ]  72.30M  3.51MB/s    eta 1m 43s 


eep-19.mat           13%[=>                  ]  73.04M  3.50MB/s    eta 1m 44s 


ep-19.mat            13%[=>                  ]  73.79M  3.50MB/s    eta 1m 44s 


p-19.mat             13%[=>                  ]  74.48M  3.47MB/s    eta 1m 44s 


-19.mat              13%[=>                  ]  75.35M  3.50MB/s    eta 1m 44s 


19.mat               13%[=>                  ]  76.13M  3.49MB/s    eta 1m 44s 


9.mat                14%[=>                  ]  76.93M  3.58MB/s    eta 1m 44s 


.mat                 14%[=>                  ]  77.71M  3.56MB/s    eta 1m 44s 


mat                  14%[=>                  ]  78.59M  3.61MB/s    eta 1m 44s 


at                   14%[=>                  ]  79.48M  3.64MB/s    eta 1m 44s 


t                    14%[=>                  ]  80.30M  3.63MB/s    eta 1m 44s 


                     14%[=>                  ]  81.12M  3.64MB/s    eta 1m 44s 


                  i  14%[=>                  ]  81.98M  3.78MB/s    eta 1m 44s 


                 im  15%[==>                 ]  82.87M  3.83MB/s    eta 1m 44s 


                ima  15%[==>                 ]  83.79M  3.90MB/s    eta 1m 44s 


               imag  15%[==>                 ]  84.68M  3.99MB/s    eta 1m 44s 


              image  15%[==>                 ]  85.55M  4.03MB/s    eta 1m 43s 


             imagen  15%[==>                 ]  86.46M  4.07MB/s    eta 1m 43s 


            imagene  15%[==>                 ]  87.40M  4.10MB/s    eta 1m 43s 


           imagenet  16%[==>                 ]  88.37M  4.17MB/s    eta 1m 43s 


          imagenet-  16%[==>                 ]  89.29M  4.23MB/s    eta 1m 43s 


         imagenet-v  16%[==>                 ]  90.35M  4.29MB/s    eta 1m 42s 


        imagenet-vg  16%[==>                 ]  91.39M  4.38MB/s    eta 1m 42s 


       imagenet-vgg  16%[==>                 ]  92.48M  4.49MB/s    eta 1m 42s 


      imagenet-vgg-  17%[==>                 ]  93.62M  4.58MB/s    eta 1m 42s 


     imagenet-vgg-v  17%[==>                 ]  94.76M  4.68MB/s    eta 1m 42s 


    imagenet-vgg-ve  17%[==>                 ]  95.95M  4.79MB/s    eta 1m 40s 


   imagenet-vgg-ver  17%[==>                 ]  97.16M  4.89MB/s    eta 1m 40s 


  imagenet-vgg-very  17%[==>                 ]  98.43M  5.01MB/s    eta 1m 40s 


 imagenet-vgg-veryd  18%[==>                 ]  99.59M  5.09MB/s    eta 1m 40s 


imagenet-vgg-veryde  18%[==>                 ] 100.65M  5.11MB/s    eta 1m 40s 


magenet-vgg-verydee  18%[==>                 ] 101.49M  5.12MB/s    eta 98s    


agenet-vgg-verydeep  18%[==>                 ] 102.32M  5.03MB/s    eta 98s    


genet-vgg-verydeep-  18%[==>                 ] 103.02M  4.98MB/s    eta 98s    


enet-vgg-verydeep-1  18%[==>                 ] 103.77M  4.90MB/s    eta 98s    


net-vgg-verydeep-19  19%[==>                 ] 104.46M  4.80MB/s    eta 98s    


et-vgg-verydeep-19.  19%[==>                 ] 105.21M  4.66MB/s    eta 98s    


t-vgg-verydeep-19.m  19%[==>                 ] 105.99M  4.56MB/s    eta 98s    


-vgg-verydeep-19.ma  19%[==>                 ] 106.88M  4.47MB/s    eta 98s    


vgg-verydeep-19.mat  19%[==>                 ] 107.54M  4.37MB/s    eta 98s    


gg-verydeep-19.mat   19%[==>                 ] 108.34M  4.29MB/s    eta 98s    


g-verydeep-19.mat    19%[==>                 ] 109.18M  4.15MB/s    eta 98s    


-verydeep-19.mat     20%[===>                ] 110.02M  4.08MB/s    eta 98s    


verydeep-19.mat      20%[===>                ] 110.80M  3.96MB/s    eta 98s    


erydeep-19.mat       20%[===>                ] 111.66M  3.89MB/s    eta 98s    


rydeep-19.mat        20%[===>                ] 112.30M  3.78MB/s    eta 98s    


ydeep-19.mat         20%[===>                ] 112.91M  3.65MB/s    eta 98s    


deep-19.mat          20%[===>                ] 113.55M  3.53MB/s    eta 98s    


eep-19.mat           20%[===>                ] 114.20M  3.53MB/s    eta 98s    


ep-19.mat            20%[===>                ] 114.85M  3.50MB/s    eta 98s    


p-19.mat             21%[===>                ] 115.54M  3.49MB/s    eta 98s    


-19.mat              21%[===>                ] 116.27M  3.52MB/s    eta 98s    


19.mat               21%[===>                ] 116.95M  3.52MB/s    eta 98s    


9.mat                21%[===>                ] 117.68M  3.51MB/s    eta 98s    


.mat                 21%[===>                ] 118.37M  3.51MB/s    eta 98s    


mat                  21%[===>                ] 118.98M  3.35MB/s    eta 98s    


at                   21%[===>                ] 119.79M  3.37MB/s    eta 99s    


t                    21%[===>                ] 120.35M  3.31MB/s    eta 99s    


                     22%[===>                ] 121.13M  3.26MB/s    eta 99s    


                  i  22%[===>                ] 121.90M  3.32MB/s    eta 99s    


                 im  22%[===>                ] 122.63M  3.24MB/s    eta 99s    


                ima  22%[===>                ] 123.34M  3.27MB/s    eta 99s    


               imag  22%[===>                ] 124.10M  3.30MB/s    eta 99s    


              image  22%[===>                ] 124.88M  3.35MB/s    eta 99s    


             imagen  22%[===>                ] 125.62M  3.34MB/s    eta 99s    


            imagene  22%[===>                ] 126.34M  3.36MB/s    eta 99s    


           imagenet  23%[===>                ] 127.09M  3.37MB/s    eta 98s    


          imagenet-  23%[===>                ] 127.88M  3.42MB/s    eta 98s    


         imagenet-v  23%[===>                ] 128.68M  3.44MB/s    eta 98s    


        imagenet-vg  23%[===>                ] 129.43M  3.48MB/s    eta 98s    


       imagenet-vgg  23%[===>                ] 130.18M  3.45MB/s    eta 98s    


      imagenet-vgg-  23%[===>                ] 130.95M  3.56MB/s    eta 98s    


     imagenet-vgg-v  23%[===>                ] 131.73M  3.59MB/s    eta 98s    


    imagenet-vgg-ve  24%[===>                ] 132.51M  3.63MB/s    eta 98s    


   imagenet-vgg-ver  24%[===>                ] 133.15M  3.63MB/s    eta 98s    


  imagenet-vgg-very  24%[===>                ] 133.96M  3.63MB/s    eta 98s    


 imagenet-vgg-veryd  24%[===>                ] 134.73M  3.62MB/s    eta 98s    


imagenet-vgg-veryde  24%[===>                ] 135.57M  3.68MB/s    eta 98s    


magenet-vgg-verydee  24%[===>                ] 136.48M  3.72MB/s    eta 98s    


agenet-vgg-verydeep  24%[===>                ] 137.32M  3.76MB/s    eta 98s    


genet-vgg-verydeep-  25%[====>               ] 138.13M  3.81MB/s    eta 98s    


enet-vgg-verydeep-1  25%[====>               ] 138.96M  3.80MB/s    eta 97s    


net-vgg-verydeep-19  25%[====>               ] 139.80M  3.84MB/s    eta 97s    


et-vgg-verydeep-19.  25%[====>               ] 140.66M  3.88MB/s    eta 97s    


t-vgg-verydeep-19.m  25%[====>               ] 141.59M  3.92MB/s    eta 97s    


-vgg-verydeep-19.ma  25%[====>               ] 142.51M  3.98MB/s    eta 97s    


vgg-verydeep-19.mat  26%[====>               ] 143.16M  3.91MB/s    eta 96s    


gg-verydeep-19.mat   26%[====>               ] 143.77M  3.90MB/s    eta 96s    


g-verydeep-19.mat    26%[====>               ] 144.48M  3.88MB/s    eta 96s    


-verydeep-19.mat     26%[====>               ] 145.15M  3.83MB/s    eta 96s    


verydeep-19.mat      26%[====>               ] 145.85M  3.81MB/s    eta 96s    


erydeep-19.mat       26%[====>               ] 146.59M  3.81MB/s    eta 96s    


rydeep-19.mat        26%[====>               ] 147.38M  3.81MB/s    eta 96s    


ydeep-19.mat         26%[====>               ] 148.20M  3.85MB/s    eta 96s    


deep-19.mat          27%[====>               ] 148.98M  3.81MB/s    eta 96s    


eep-19.mat           27%[====>               ] 149.62M  3.74MB/s    eta 96s    


ep-19.mat            27%[====>               ] 150.16M  3.69MB/s    eta 95s    


p-19.mat             27%[====>               ] 150.95M  3.65MB/s    eta 95s    


-19.mat              27%[====>               ] 151.49M  3.60MB/s    eta 95s    


19.mat               27%[====>               ] 152.16M  3.53MB/s    eta 95s    


9.mat                27%[====>               ] 152.65M  3.47MB/s    eta 95s    


.mat                 27%[====>               ] 153.35M  3.39MB/s    eta 95s    


mat                  28%[====>               ] 154.05M  3.38MB/s    eta 95s    


at                   28%[====>               ] 154.70M  3.28MB/s    eta 95s    


t                    28%[====>               ] 155.37M  3.31MB/s    eta 95s    


                     28%[====>               ] 156.10M  3.32MB/s    eta 95s    


                  i  28%[====>               ] 156.84M  3.34MB/s    eta 95s    


                 im  28%[====>               ] 157.51M  3.30MB/s    eta 95s    


                ima  28%[====>               ] 158.21M  3.30MB/s    eta 95s    


               imag  28%[====>               ] 158.90M  3.29MB/s    eta 95s    


              image  29%[====>               ] 159.60M  3.28MB/s    eta 95s    


             imagen  29%[====>               ] 160.30M  3.24MB/s    eta 94s    


            imagene  29%[====>               ] 161.05M  3.26MB/s    eta 94s    


           imagenet  29%[====>               ] 161.77M  3.31MB/s    eta 94s    


          imagenet-  29%[====>               ] 162.48M  3.28MB/s    eta 94s    


         imagenet-v  29%[====>               ] 163.04M  3.20MB/s    eta 94s    


        imagenet-vg  29%[====>               ] 163.84M  3.28MB/s    eta 94s    


       imagenet-vgg  29%[====>               ] 164.41M  3.28MB/s    eta 94s    


      imagenet-vgg-  30%[=====>              ] 165.15M  3.26MB/s    eta 94s    


     imagenet-vgg-v  30%[=====>              ] 165.95M  3.31MB/s    eta 94s    


    imagenet-vgg-ve  30%[=====>              ] 166.65M  3.29MB/s    eta 94s    


   imagenet-vgg-ver  30%[=====>              ] 167.43M  3.36MB/s    eta 94s    


  imagenet-vgg-very  30%[=====>              ] 168.20M  3.33MB/s    eta 94s    


 imagenet-vgg-veryd  30%[=====>              ] 169.01M  3.41MB/s    eta 94s    


imagenet-vgg-veryde  30%[=====>              ] 169.76M  3.38MB/s    eta 94s    


magenet-vgg-verydee  31%[=====>              ] 170.55M  3.43MB/s    eta 94s    


agenet-vgg-verydeep  31%[=====>              ] 171.35M  3.43MB/s    eta 93s    


genet-vgg-verydeep-  31%[=====>              ] 172.13M  3.48MB/s    eta 93s    


enet-vgg-verydeep-1  31%[=====>              ] 172.93M  3.49MB/s    eta 93s    


net-vgg-verydeep-19  31%[=====>              ] 173.74M  3.53MB/s    eta 93s    


et-vgg-verydeep-19.  31%[=====>              ] 174.51M  3.51MB/s    eta 93s    


t-vgg-verydeep-19.m  31%[=====>              ] 175.35M  3.63MB/s    eta 92s    


-vgg-verydeep-19.ma  32%[=====>              ] 176.20M  3.71MB/s    eta 92s    


vgg-verydeep-19.mat  32%[=====>              ] 177.01M  3.77MB/s    eta 92s    


gg-verydeep-19.mat   32%[=====>              ] 177.82M  3.80MB/s    eta 92s    


g-verydeep-19.mat    32%[=====>              ] 178.70M  3.87MB/s    eta 92s    


-verydeep-19.mat     32%[=====>              ] 179.52M  3.89MB/s    eta 91s    


verydeep-19.mat      32%[=====>              ] 180.35M  3.88MB/s    eta 91s    


erydeep-19.mat       32%[=====>              ] 181.04M  3.87MB/s    eta 91s    


rydeep-19.mat        33%[=====>              ] 181.87M  3.88MB/s    eta 91s    


ydeep-19.mat         33%[=====>              ] 182.51M  3.86MB/s    eta 91s    


deep-19.mat          33%[=====>              ] 183.01M  3.74MB/s    eta 91s    


eep-19.mat           33%[=====>              ] 183.76M  3.75MB/s    eta 91s    


ep-19.mat            33%[=====>              ] 184.51M  3.69MB/s    eta 91s    


p-19.mat             33%[=====>              ] 185.26M  3.67MB/s    eta 91s    


-19.mat              33%[=====>              ] 186.02M  3.68MB/s    eta 91s    


19.mat               33%[=====>              ] 186.77M  3.66MB/s    eta 90s    


9.mat                34%[=====>              ] 187.59M  3.64MB/s    eta 90s    


.mat                 34%[=====>              ] 188.37M  3.63MB/s    eta 90s    


mat                  34%[=====>              ] 189.23M  3.63MB/s    eta 90s    


at                   34%[=====>              ] 190.05M  3.65MB/s    eta 90s    


t                    34%[=====>              ] 190.90M  3.62MB/s    eta 89s    


                     34%[=====>              ] 191.71M  3.61MB/s    eta 89s    


                  i  35%[======>             ] 192.55M  3.61MB/s    eta 89s    


                 im  35%[======>             ] 193.41M  3.63MB/s    eta 89s    


                ima  35%[======>             ] 194.26M  3.71MB/s    eta 89s    


               imag  35%[======>             ] 195.13M  3.74MB/s    eta 88s    


              image  35%[======>             ] 195.82M  3.78MB/s    eta 88s    


             imagen  35%[======>             ] 196.44M  3.77MB/s    eta 88s    


            imagene  35%[======>             ] 197.27M  3.82MB/s    eta 88s    


           imagenet  36%[======>             ] 198.10M  3.86MB/s    eta 88s    


          imagenet-  36%[======>             ] 199.02M  3.88MB/s    eta 87s    


         imagenet-v  36%[======>             ] 199.95M  3.97MB/s    eta 87s    


        imagenet-vg  36%[======>             ] 200.80M  4.00MB/s    eta 87s    


       imagenet-vgg  36%[======>             ] 201.65M  4.02MB/s    eta 87s    


      imagenet-vgg-  36%[======>             ] 202.47M  3.99MB/s    eta 87s    


     imagenet-vgg-v  37%[======>             ] 203.30M  4.01MB/s    eta 86s    


    imagenet-vgg-ve  37%[======>             ] 204.18M  4.00MB/s    eta 86s    


   imagenet-vgg-ver  37%[======>             ] 205.07M  4.05MB/s    eta 86s    


  imagenet-vgg-very  37%[======>             ] 205.95M  4.04MB/s    eta 86s    


 imagenet-vgg-veryd  37%[======>             ] 206.79M  4.03MB/s    eta 86s    


imagenet-vgg-veryde  37%[======>             ] 207.62M  4.06MB/s    eta 85s    


magenet-vgg-verydee  37%[======>             ] 208.52M  4.04MB/s    eta 85s    


agenet-vgg-verydeep  38%[======>             ] 209.37M  4.14MB/s    eta 85s    


genet-vgg-verydeep-  38%[======>             ] 210.23M  4.18MB/s    eta 85s    


enet-vgg-verydeep-1  38%[======>             ] 211.12M  4.17MB/s    eta 85s    


net-vgg-verydeep-19  38%[======>             ] 211.98M  4.24MB/s    eta 84s    


et-vgg-verydeep-19.  38%[======>             ] 212.84M  4.20MB/s    eta 84s    


t-vgg-verydeep-19.m  38%[======>             ] 213.73M  4.25MB/s    eta 84s    


-vgg-verydeep-19.ma  39%[======>             ] 214.60M  4.24MB/s    eta 84s    


vgg-verydeep-19.mat  39%[======>             ] 215.46M  4.18MB/s    eta 84s    


gg-verydeep-19.mat   39%[======>             ] 216.09M  4.12MB/s    eta 83s    


g-verydeep-19.mat    39%[======>             ] 216.71M  4.00MB/s    eta 83s    


-verydeep-19.mat     39%[======>             ] 217.54M  3.94MB/s    eta 83s    


verydeep-19.mat      39%[======>             ] 218.30M  3.91MB/s    eta 83s    


erydeep-19.mat       39%[======>             ] 219.02M  3.89MB/s    eta 83s    


rydeep-19.mat        40%[=======>            ] 219.88M  3.89MB/s    eta 82s    


ydeep-19.mat         40%[=======>            ] 220.68M  3.84MB/s    eta 82s    


deep-19.mat          40%[=======>            ] 221.48M  3.83MB/s    eta 82s    


eep-19.mat           40%[=======>            ] 222.30M  3.81MB/s    eta 82s    


ep-19.mat            40%[=======>            ] 223.13M  3.77MB/s    eta 82s    


p-19.mat             40%[=======>            ] 224.04M  3.80MB/s    eta 81s    


-19.mat              40%[=======>            ] 224.98M  3.81MB/s    eta 81s    


19.mat               41%[=======>            ] 225.84M  3.79MB/s    eta 81s    


9.mat                41%[=======>            ] 226.73M  3.80MB/s    eta 81s    


.mat                 41%[=======>            ] 227.60M  3.81MB/s    eta 81s    


mat                  41%[=======>            ] 228.49M  3.88MB/s    eta 80s    


at                   41%[=======>            ] 229.38M  3.96MB/s    eta 80s    


t                    41%[=======>            ] 229.98M  3.89MB/s    eta 80s    


                     42%[=======>            ] 230.90M  3.97MB/s    eta 80s    


                  i  42%[=======>            ] 231.52M  3.92MB/s    eta 80s    


                 im  42%[=======>            ] 232.20M  3.89MB/s    eta 79s    


                ima  42%[=======>            ] 232.87M  3.86MB/s    eta 79s    


               imag  42%[=======>            ] 233.60M  3.84MB/s    eta 79s    


              image  42%[=======>            ] 234.30M  3.81MB/s    eta 79s    


             imagen  42%[=======>            ] 234.99M  3.76MB/s    eta 79s    


            imagene  42%[=======>            ] 235.70M  3.74MB/s    eta 78s    


           imagenet  43%[=======>            ] 236.46M  3.70MB/s    eta 78s    


          imagenet-  43%[=======>            ] 237.21M  3.62MB/s    eta 78s    


         imagenet-v  43%[=======>            ] 238.01M  3.63MB/s    eta 78s    


        imagenet-vg  43%[=======>            ] 238.77M  3.59MB/s    eta 78s    


       imagenet-vgg  43%[=======>            ] 239.51M  3.51MB/s    eta 77s    


      imagenet-vgg-  43%[=======>            ] 240.34M  3.49MB/s    eta 77s    


     imagenet-vgg-v  43%[=======>            ] 241.12M  3.53MB/s    eta 77s    


    imagenet-vgg-ve  44%[=======>            ] 241.95M  3.60MB/s    eta 77s    


   imagenet-vgg-ver  44%[=======>            ] 242.71M  3.55MB/s    eta 77s    


  imagenet-vgg-very  44%[=======>            ] 243.51M  3.61MB/s    eta 77s    


 imagenet-vgg-veryd  44%[=======>            ] 244.27M  3.63MB/s    eta 77s    


imagenet-vgg-veryde  44%[=======>            ] 245.09M  3.69MB/s    eta 77s    


magenet-vgg-verydee  44%[=======>            ] 245.70M  3.62MB/s    eta 77s    


agenet-vgg-verydeep  44%[=======>            ] 246.29M  3.57MB/s    eta 77s    


genet-vgg-verydeep-  44%[=======>            ] 247.07M  3.61MB/s    eta 76s    


enet-vgg-verydeep-1  45%[========>           ] 247.77M  3.59MB/s    eta 76s    


net-vgg-verydeep-19  45%[========>           ] 248.54M  3.60MB/s    eta 76s    


et-vgg-verydeep-19.  45%[========>           ] 249.35M  3.62MB/s    eta 76s    


t-vgg-verydeep-19.m  45%[========>           ] 250.24M  3.66MB/s    eta 76s    


-vgg-verydeep-19.ma  45%[========>           ] 251.09M  3.70MB/s    eta 75s    


vgg-verydeep-19.mat  45%[========>           ] 251.90M  3.66MB/s    eta 75s    


gg-verydeep-19.mat   46%[========>           ] 252.71M  3.66MB/s    eta 75s    


g-verydeep-19.mat    46%[========>           ] 253.57M  3.73MB/s    eta 75s    


-verydeep-19.mat     46%[========>           ] 254.40M  3.71MB/s    eta 75s    


verydeep-19.mat      46%[========>           ] 255.34M  3.80MB/s    eta 74s    


erydeep-19.mat       46%[========>           ] 256.15M  3.75MB/s    eta 74s    


rydeep-19.mat        46%[========>           ] 256.90M  3.72MB/s    eta 74s    


ydeep-19.mat         46%[========>           ] 257.35M  3.65MB/s    eta 74s    


deep-19.mat          46%[========>           ] 257.79M  3.58MB/s    eta 74s    


eep-19.mat           47%[========>           ] 258.24M  3.53MB/s    eta 73s    

ep-19.mat            47%[========>           ] 258.74M  3.47MB/s    eta 73s    


p-19.mat             47%[========>           ] 259.20M  3.41MB/s    eta 73s    


-19.mat              47%[========>           ] 259.60M  3.26MB/s    eta 73s    


19.mat               47%[========>           ] 260.18M  3.20MB/s    eta 73s    


9.mat                47%[========>           ] 260.68M  3.12MB/s    eta 73s    


.mat                 47%[========>           ] 261.18M  2.96MB/s    eta 73s    


mat                  47%[========>           ] 261.73M  2.86MB/s    eta 73s    


at                   47%[========>           ] 262.27M  2.82MB/s    eta 73s    


t                    47%[========>           ] 262.80M  2.65MB/s    eta 73s    


                     47%[========>           ] 263.32M  2.56MB/s    eta 73s    


                  i  48%[========>           ] 263.90M  2.51MB/s    eta 73s    


                 im  48%[========>           ] 264.40M  2.43MB/s    eta 73s    


                ima  48%[========>           ] 264.91M  2.40MB/s    eta 73s    


               imag  48%[========>           ] 265.45M  2.39MB/s    eta 73s    


              image  48%[========>           ] 265.99M  2.43MB/s    eta 73s    


             imagen  48%[========>           ] 266.59M  2.48MB/s    eta 73s    


            imagene  48%[========>           ] 267.07M  2.49MB/s    eta 73s    


           imagenet  48%[========>           ] 267.70M  2.53MB/s    eta 73s    


          imagenet-  48%[========>           ] 268.26M  2.58MB/s    eta 73s    


         imagenet-v  48%[========>           ] 268.84M  2.60MB/s    eta 72s    


        imagenet-vg  49%[========>           ] 269.41M  2.59MB/s    eta 72s    


       imagenet-vgg  49%[========>           ] 269.99M  2.64MB/s    eta 72s    


      imagenet-vgg-  49%[========>           ] 270.62M  2.68MB/s    eta 72s    


     imagenet-vgg-v  49%[========>           ] 271.15M  2.66MB/s    eta 72s    


    imagenet-vgg-ve  49%[========>           ] 271.70M  2.70MB/s    eta 72s    


   imagenet-vgg-ver  49%[========>           ] 272.29M  2.71MB/s    eta 72s    


  imagenet-vgg-very  49%[========>           ] 272.90M  2.70MB/s    eta 72s    


 imagenet-vgg-veryd  49%[========>           ] 273.57M  2.76MB/s    eta 72s    


imagenet-vgg-veryde  49%[========>           ] 274.12M  2.76MB/s    eta 72s    


magenet-vgg-verydee  50%[=========>          ] 274.82M  2.81MB/s    eta 71s    


agenet-vgg-verydeep  50%[=========>          ] 275.48M  2.84MB/s    eta 71s    


genet-vgg-verydeep-  50%[=========>          ] 276.16M  2.89MB/s    eta 71s    


enet-vgg-verydeep-1  50%[=========>          ] 276.80M  2.91MB/s    eta 71s    


net-vgg-verydeep-19  50%[=========>          ] 277.48M  2.95MB/s    eta 71s    


et-vgg-verydeep-19.  50%[=========>          ] 278.24M  3.00MB/s    eta 71s    


t-vgg-verydeep-19.m  50%[=========>          ] 278.91M  3.01MB/s    eta 71s    


-vgg-verydeep-19.ma  50%[=========>          ] 279.63M  3.05MB/s    eta 71s    


vgg-verydeep-19.mat  51%[=========>          ] 280.34M  3.07MB/s    eta 71s    


gg-verydeep-19.mat   51%[=========>          ] 281.09M  3.12MB/s    eta 71s    


g-verydeep-19.mat    51%[=========>          ] 281.85M  3.18MB/s    eta 70s    


-verydeep-19.mat     51%[=========>          ] 282.63M  3.24MB/s    eta 70s    


verydeep-19.mat      51%[=========>          ] 283.40M  3.32MB/s    eta 70s    


erydeep-19.mat       51%[=========>          ] 284.21M  3.41MB/s    eta 70s    


rydeep-19.mat        51%[=========>          ] 285.06M  3.49MB/s    eta 70s    


ydeep-19.mat         52%[=========>          ] 285.98M  3.57MB/s    eta 69s    


deep-19.mat          52%[=========>          ] 286.95M  3.67MB/s    eta 69s    


eep-19.mat           52%[=========>          ] 287.95M  3.76MB/s    eta 69s    


ep-19.mat            52%[=========>          ] 288.91M  3.81MB/s    eta 69s    


p-19.mat             52%[=========>          ] 289.82M  3.87MB/s    eta 69s    


-19.mat              52%[=========>          ] 290.43M  3.83MB/s    eta 67s    


19.mat               53%[=========>          ] 291.23M  3.85MB/s    eta 67s    


9.mat                53%[=========>          ] 292.04M  3.86MB/s    eta 67s    


.mat                 53%[=========>          ] 292.93M  3.92MB/s    eta 67s    


mat                  53%[=========>          ] 293.82M  3.99MB/s    eta 67s    


at                   53%[=========>          ] 294.77M  4.06MB/s    eta 66s    


t                    53%[=========>          ] 295.63M  4.09MB/s    eta 66s    


                     53%[=========>          ] 296.54M  4.14MB/s    eta 66s    


                  i  54%[=========>          ] 297.59M  4.23MB/s    eta 66s    


                 im  54%[=========>          ] 298.57M  4.27MB/s    eta 66s    


                ima  54%[=========>          ] 299.55M  4.28MB/s    eta 65s    


               imag  54%[=========>          ] 300.54M  4.30MB/s    eta 65s    


              image  54%[=========>          ] 301.57M  4.32MB/s    eta 65s    


             imagen  55%[==========>         ] 302.59M  4.34MB/s    eta 65s    


            imagene  55%[==========>         ] 303.59M  4.37MB/s    eta 65s    


           imagenet  55%[==========>         ] 304.62M  4.48MB/s    eta 63s    


          imagenet-  55%[==========>         ] 305.59M  4.57MB/s    eta 63s    


         imagenet-v  55%[==========>         ] 306.62M  4.66MB/s    eta 63s    


        imagenet-vg  56%[==========>         ] 307.70M  4.82MB/s    eta 63s    


       imagenet-vgg  56%[==========>         ] 308.46M  4.72MB/s    eta 63s    


      imagenet-vgg-  56%[==========>         ] 309.52M  4.81MB/s    eta 62s    


     imagenet-vgg-v  56%[==========>         ] 310.31M  4.75MB/s    eta 62s    


    imagenet-vgg-ve  56%[==========>         ] 310.95M  4.70MB/s    eta 62s    


   imagenet-vgg-ver  56%[==========>         ] 311.63M  4.61MB/s    eta 62s    


  imagenet-vgg-very  56%[==========>         ] 312.55M  4.62MB/s    eta 62s    


 imagenet-vgg-veryd  57%[==========>         ] 313.41M  4.56MB/s    eta 61s    


imagenet-vgg-veryde  57%[==========>         ] 314.30M  4.51MB/s    eta 61s    


magenet-vgg-verydee  57%[==========>         ] 315.24M  4.48MB/s    eta 61s    


agenet-vgg-verydeep  57%[==========>         ] 316.20M  4.49MB/s    eta 61s    


genet-vgg-verydeep-  57%[==========>         ] 317.16M  4.47MB/s    eta 61s    


enet-vgg-verydeep-1  57%[==========>         ] 318.18M  4.48MB/s    eta 59s    


net-vgg-verydeep-19  58%[==========>         ] 319.13M  4.43MB/s    eta 59s    


et-vgg-verydeep-19.  58%[==========>         ] 320.10M  4.44MB/s    eta 59s    


t-vgg-verydeep-19.m  58%[==========>         ] 321.15M  4.39MB/s    eta 59s    


-vgg-verydeep-19.ma  58%[==========>         ] 322.13M  4.41MB/s    eta 59s    


vgg-verydeep-19.mat  58%[==========>         ] 323.16M  4.41MB/s    eta 58s    


gg-verydeep-19.mat   59%[==========>         ] 324.26M  4.53MB/s    eta 58s    


g-verydeep-19.mat    59%[==========>         ] 325.29M  4.62MB/s    eta 58s    


-verydeep-19.mat     59%[==========>         ] 326.32M  4.74MB/s    eta 58s    


verydeep-19.mat      59%[==========>         ] 327.34M  4.82MB/s    eta 58s    


erydeep-19.mat       59%[==========>         ] 328.40M  4.85MB/s    eta 56s    


rydeep-19.mat        59%[==========>         ] 329.26M  4.85MB/s    eta 56s    


ydeep-19.mat         60%[===========>        ] 329.96M  4.77MB/s    eta 56s    


deep-19.mat          60%[===========>        ] 330.87M  4.79MB/s    eta 56s    


eep-19.mat           60%[===========>        ] 331.54M  4.63MB/s    eta 56s    


ep-19.mat            60%[===========>        ] 332.29M  4.56MB/s    eta 55s    


p-19.mat             60%[===========>        ] 332.87M  4.42MB/s    eta 55s    


-19.mat              60%[===========>        ] 333.48M  4.30MB/s    eta 55s    


19.mat               60%[===========>        ] 334.09M  4.18MB/s    eta 55s    


9.mat                60%[===========>        ] 334.73M  4.06MB/s    eta 55s    


.mat                 61%[===========>        ] 335.38M  3.94MB/s    eta 55s    


mat                  61%[===========>        ] 336.04M  3.80MB/s    eta 55s    


at                   61%[===========>        ] 336.68M  3.70MB/s    eta 55s    


t                    61%[===========>        ] 337.30M  3.62MB/s    eta 55s    


                     61%[===========>        ] 338.02M  3.54MB/s    eta 55s    


                  i  61%[===========>        ] 338.66M  3.36MB/s    eta 54s    


                 im  61%[===========>        ] 339.24M  3.26MB/s    eta 54s    


                ima  61%[===========>        ] 339.95M  3.21MB/s    eta 54s    


               imag  62%[===========>        ] 340.62M  3.16MB/s    eta 54s    


              image  62%[===========>        ] 341.30M  3.07MB/s    eta 54s    


             imagen  62%[===========>        ] 341.99M  3.12MB/s    eta 53s    


            imagene  62%[===========>        ] 342.66M  3.11MB/s    eta 53s    


           imagenet  62%[===========>        ] 343.34M  3.15MB/s    eta 53s    


          imagenet-  62%[===========>        ] 343.98M  3.16MB/s    eta 53s    


         imagenet-v  62%[===========>        ] 344.59M  3.15MB/s    eta 53s    


        imagenet-vg  62%[===========>        ] 345.24M  3.14MB/s    eta 53s    


       imagenet-vgg  62%[===========>        ] 345.91M  3.13MB/s    eta 53s    


      imagenet-vgg-  63%[===========>        ] 346.63M  3.13MB/s    eta 53s    


     imagenet-vgg-v  63%[===========>        ] 347.37M  3.16MB/s    eta 53s    


    imagenet-vgg-ve  63%[===========>        ] 348.16M  3.19MB/s    eta 53s    


   imagenet-vgg-ver  63%[===========>        ] 348.90M  3.20MB/s    eta 52s    


  imagenet-vgg-very  63%[===========>        ] 349.60M  3.21MB/s    eta 52s    


 imagenet-vgg-veryd  63%[===========>        ] 350.38M  3.25MB/s    eta 52s    


imagenet-vgg-veryde  63%[===========>        ] 351.20M  3.30MB/s    eta 52s    


magenet-vgg-verydee  64%[===========>        ] 351.93M  3.33MB/s    eta 52s    


agenet-vgg-verydeep  64%[===========>        ] 352.70M  3.32MB/s    eta 51s    


genet-vgg-verydeep-  64%[===========>        ] 353.51M  3.37MB/s    eta 51s    


enet-vgg-verydeep-1  64%[===========>        ] 354.35M  3.42MB/s    eta 51s    


net-vgg-verydeep-19  64%[===========>        ] 355.15M  3.41MB/s    eta 51s    


et-vgg-verydeep-19.  64%[===========>        ] 355.91M  3.45MB/s    eta 51s    


t-vgg-verydeep-19.m  64%[===========>        ] 356.69M  3.51MB/s    eta 50s    


-vgg-verydeep-19.ma  65%[============>       ] 357.54M  3.60MB/s    eta 50s    


vgg-verydeep-19.mat  65%[============>       ] 358.29M  3.66MB/s    eta 50s    


gg-verydeep-19.mat   65%[============>       ] 359.10M  3.70MB/s    eta 50s    


g-verydeep-19.mat    65%[============>       ] 359.84M  3.72MB/s    eta 50s    


-verydeep-19.mat     65%[============>       ] 360.37M  3.65MB/s    eta 49s    


verydeep-19.mat      65%[============>       ] 361.16M  3.68MB/s    eta 49s    


erydeep-19.mat       65%[============>       ] 361.73M  3.64MB/s    eta 49s    


rydeep-19.mat        65%[============>       ] 362.44M  3.61MB/s    eta 49s    


ydeep-19.mat         66%[============>       ] 363.13M  3.58MB/s    eta 49s    


deep-19.mat          66%[============>       ] 363.79M  3.52MB/s    eta 48s    


eep-19.mat           66%[============>       ] 364.51M  3.55MB/s    eta 48s    


ep-19.mat            66%[============>       ] 365.23M  3.53MB/s    eta 48s    


p-19.mat             66%[============>       ] 365.95M  3.49MB/s    eta 48s    


-19.mat              66%[============>       ] 366.66M  3.51MB/s    eta 48s    


19.mat               66%[============>       ] 367.43M  3.48MB/s    eta 47s    


9.mat                67%[============>       ] 368.23M  3.45MB/s    eta 47s    


.mat                 67%[============>       ] 368.99M  3.48MB/s    eta 47s    


mat                  67%[============>       ] 369.77M  3.49MB/s    eta 47s    


at                   67%[============>       ] 370.60M  3.49MB/s    eta 47s    


t                    67%[============>       ] 371.40M  3.47MB/s    eta 46s    


                     67%[============>       ] 372.24M  3.56MB/s    eta 46s    


                  i  67%[============>       ] 373.05M  3.58MB/s    eta 46s    


                 im  68%[============>       ] 373.87M  3.65MB/s    eta 46s    


                ima  68%[============>       ] 374.55M  3.66MB/s    eta 46s    


               imag  68%[============>       ] 375.27M  3.67MB/s    eta 45s    


              image  68%[============>       ] 375.98M  3.70MB/s    eta 45s    


             imagen  68%[============>       ] 376.45M  3.62MB/s    eta 45s    


            imagene  68%[============>       ] 377.16M  3.61MB/s    eta 45s    


           imagenet  68%[============>       ] 377.87M  3.61MB/s    eta 45s    


          imagenet-  68%[============>       ] 378.52M  3.59MB/s    eta 44s    


         imagenet-v  69%[============>       ] 379.20M  3.53MB/s    eta 44s    


        imagenet-vg  69%[============>       ] 379.90M  3.52MB/s    eta 44s    


       imagenet-vgg  69%[============>       ] 380.65M  3.51MB/s    eta 44s    


      imagenet-vgg-  69%[============>       ] 381.32M  3.47MB/s    eta 44s    


     imagenet-vgg-v  69%[============>       ] 382.05M  3.46MB/s    eta 44s    


    imagenet-vgg-ve  69%[============>       ] 382.84M  3.45MB/s    eta 44s    


   imagenet-vgg-ver  69%[============>       ] 383.63M  3.47MB/s    eta 44s    


  imagenet-vgg-very  69%[============>       ] 384.43M  3.45MB/s    eta 44s    


 imagenet-vgg-veryd  70%[=============>      ] 385.16M  3.45MB/s    eta 44s    


imagenet-vgg-veryde  70%[=============>      ] 385.98M  3.45MB/s    eta 43s    


magenet-vgg-verydee  70%[=============>      ] 386.79M  3.53MB/s    eta 43s    


agenet-vgg-verydeep  70%[=============>      ] 387.54M  3.55MB/s    eta 43s    


genet-vgg-verydeep-  70%[=============>      ] 388.32M  3.64MB/s    eta 43s    


enet-vgg-verydeep-1  70%[=============>      ] 388.99M  3.63MB/s    eta 43s    


net-vgg-verydeep-19  70%[=============>      ] 389.79M  3.66MB/s    eta 42s    


et-vgg-verydeep-19.  71%[=============>      ] 390.71M  3.75MB/s    eta 42s    


t-vgg-verydeep-19.m  71%[=============>      ] 391.48M  3.77MB/s    eta 42s    


-vgg-verydeep-19.ma  71%[=============>      ] 392.40M  3.81MB/s    eta 42s    


vgg-verydeep-19.mat  71%[=============>      ] 393.16M  3.85MB/s    eta 42s    


gg-verydeep-19.mat   71%[=============>      ] 394.02M  3.87MB/s    eta 40s    


g-verydeep-19.mat    71%[=============>      ] 394.85M  3.91MB/s    eta 40s    


-verydeep-19.mat     72%[=============>      ] 395.71M  3.92MB/s    eta 40s    


verydeep-19.mat      72%[=============>      ] 396.48M  3.95MB/s    eta 40s    


erydeep-19.mat       72%[=============>      ] 397.32M  3.96MB/s    eta 40s    


rydeep-19.mat        72%[=============>      ] 398.13M  3.92MB/s    eta 39s    


ydeep-19.mat         72%[=============>      ] 398.96M  3.97MB/s    eta 39s    


deep-19.mat          72%[=============>      ] 399.73M  3.93MB/s    eta 39s    


eep-19.mat           72%[=============>      ] 400.59M  3.96MB/s    eta 39s    


ep-19.mat            73%[=============>      ] 401.38M  3.99MB/s    eta 39s    


p-19.mat             73%[=============>      ] 402.23M  4.03MB/s    eta 38s    


-19.mat              73%[=============>      ] 403.01M  4.01MB/s    eta 38s    


19.mat               73%[=============>      ] 403.80M  4.02MB/s    eta 38s    


9.mat                73%[=============>      ] 404.65M  4.04MB/s    eta 38s    


.mat                 73%[=============>      ] 405.45M  4.04MB/s    eta 38s    


mat                  73%[=============>      ] 406.27M  4.02MB/s    eta 37s    


at                   74%[=============>      ] 407.21M  4.06MB/s    eta 37s    


t                    74%[=============>      ] 408.12M  4.10MB/s    eta 37s    


                     74%[=============>      ] 408.87M  4.03MB/s    eta 37s    


                  i  74%[=============>      ] 409.55M  3.95MB/s    eta 37s    


                 im  74%[=============>      ] 410.54M  3.97MB/s    eta 36s    


                ima  74%[=============>      ] 411.18M  3.88MB/s    eta 36s    


               imag  74%[=============>      ] 411.93M  3.86MB/s    eta 36s    


              image  75%[==============>     ] 412.71M  3.88MB/s    eta 36s    


             imagen  75%[==============>     ] 413.48M  3.81MB/s    eta 36s    

            imagene  75%[==============>     ] 414.23M  3.78MB/s    eta 35s    


           imagenet  75%[==============>     ] 414.96M  3.78MB/s    eta 35s    


          imagenet-  75%[==============>     ] 415.74M  3.74MB/s    eta 35s    


         imagenet-v  75%[==============>     ] 416.63M  3.79MB/s    eta 35s    


        imagenet-vg  75%[==============>     ] 417.34M  3.74MB/s    eta 35s    


       imagenet-vgg  76%[==============>     ] 418.12M  3.73MB/s    eta 34s    


      imagenet-vgg-  76%[==============>     ] 418.73M  3.70MB/s    eta 34s    


     imagenet-vgg-v  76%[==============>     ] 419.35M  3.62MB/s    eta 34s    


    imagenet-vgg-ve  76%[==============>     ] 419.95M  3.59MB/s    eta 34s    


   imagenet-vgg-ver  76%[==============>     ] 420.51M  3.51MB/s    eta 34s    


  imagenet-vgg-very  76%[==============>     ] 421.15M  3.39MB/s    eta 33s    


 imagenet-vgg-veryd  76%[==============>     ] 421.80M  3.34MB/s    eta 33s    


imagenet-vgg-veryde  76%[==============>     ] 422.52M  3.33MB/s    eta 33s    


magenet-vgg-verydee  77%[==============>     ] 423.37M  3.31MB/s    eta 33s    


agenet-vgg-verydeep  77%[==============>     ] 424.02M  3.35MB/s    eta 33s    


genet-vgg-verydeep-  77%[==============>     ] 424.66M  3.34MB/s    eta 33s    


enet-vgg-verydeep-1  77%[==============>     ] 425.32M  3.30MB/s    eta 33s    


net-vgg-verydeep-19  77%[==============>     ] 426.04M  3.33MB/s    eta 33s    


et-vgg-verydeep-19.  77%[==============>     ] 426.79M  3.31MB/s    eta 33s    


t-vgg-verydeep-19.m  77%[==============>     ] 427.49M  3.26MB/s    eta 33s    


-vgg-verydeep-19.ma  77%[==============>     ] 428.26M  3.26MB/s    eta 32s    


vgg-verydeep-19.mat  78%[==============>     ] 429.05M  3.26MB/s    eta 32s    


gg-verydeep-19.mat   78%[==============>     ] 429.76M  3.20MB/s    eta 32s    


g-verydeep-19.mat    78%[==============>     ] 430.38M  3.18MB/s    eta 32s    


-verydeep-19.mat     78%[==============>     ] 431.07M  3.21MB/s    eta 32s    


verydeep-19.mat      78%[==============>     ] 431.76M  3.22MB/s    eta 31s    


erydeep-19.mat       78%[==============>     ] 432.46M  3.25MB/s    eta 31s    


rydeep-19.mat        78%[==============>     ] 433.20M  3.36MB/s    eta 31s    


ydeep-19.mat         78%[==============>     ] 433.91M  3.38MB/s    eta 31s    


deep-19.mat          79%[==============>     ] 434.60M  3.43MB/s    eta 31s    


eep-19.mat           79%[==============>     ] 435.32M  3.44MB/s    eta 30s    


ep-19.mat            79%[==============>     ] 435.88M  3.38MB/s    eta 30s    


p-19.mat             79%[==============>     ] 436.41M  3.37MB/s    eta 30s    


-19.mat              79%[==============>     ] 437.15M  3.35MB/s    eta 30s    


19.mat               79%[==============>     ] 437.73M  3.34MB/s    eta 30s    


9.mat                79%[==============>     ] 438.41M  3.34MB/s    eta 29s    


.mat                 79%[==============>     ] 439.12M  3.29MB/s    eta 29s    


mat                  80%[===============>    ] 439.87M  3.31MB/s    eta 29s    


at                   80%[===============>    ] 440.55M  3.30MB/s    eta 29s    


t                    80%[===============>    ] 441.27M  3.28MB/s    eta 29s    


                     80%[===============>    ] 442.01M  3.35MB/s    eta 28s    


                  i  80%[===============>    ] 442.73M  3.32MB/s    eta 28s    


                 im  80%[===============>    ] 443.43M  3.36MB/s    eta 28s    


                ima  80%[===============>    ] 444.12M  3.36MB/s    eta 28s    


               imag  80%[===============>    ] 444.82M  3.30MB/s    eta 28s    


              image  81%[===============>    ] 445.55M  3.32MB/s    eta 27s    


             imagen  81%[===============>    ] 446.27M  3.31MB/s    eta 27s    


            imagene  81%[===============>    ] 446.98M  3.33MB/s    eta 27s    


           imagenet  81%[===============>    ] 447.70M  3.41MB/s    eta 27s    


          imagenet-  81%[===============>    ] 448.27M  3.36MB/s    eta 27s    


         imagenet-v  81%[===============>    ] 448.90M  3.32MB/s    eta 26s    


        imagenet-vg  81%[===============>    ] 449.66M  3.35MB/s    eta 26s    


       imagenet-vgg  81%[===============>    ] 450.20M  3.24MB/s    eta 26s    


      imagenet-vgg-  82%[===============>    ] 450.76M  3.22MB/s    eta 26s    


     imagenet-vgg-v  82%[===============>    ] 451.29M  3.16MB/s    eta 26s    


    imagenet-vgg-ve  82%[===============>    ] 451.91M  3.13MB/s    eta 26s    


   imagenet-vgg-ver  82%[===============>    ] 452.60M  3.14MB/s    eta 26s    


  imagenet-vgg-very  82%[===============>    ] 453.23M  3.09MB/s    eta 26s    


 imagenet-vgg-veryd  82%[===============>    ] 453.87M  3.08MB/s    eta 26s    


imagenet-vgg-veryde  82%[===============>    ] 454.51M  3.03MB/s    eta 26s    


magenet-vgg-verydee  82%[===============>    ] 455.15M  3.00MB/s    eta 25s    


agenet-vgg-verydeep  82%[===============>    ] 455.82M  2.99MB/s    eta 25s    


genet-vgg-verydeep-  83%[===============>    ] 456.51M  2.97MB/s    eta 25s    


enet-vgg-verydeep-1  83%[===============>    ] 457.26M  2.96MB/s    eta 25s    


net-vgg-verydeep-19  83%[===============>    ] 457.96M  2.98MB/s    eta 25s    


et-vgg-verydeep-19.  83%[===============>    ] 458.63M  2.98MB/s    eta 24s    


t-vgg-verydeep-19.m  83%[===============>    ] 459.35M  3.05MB/s    eta 24s    


-vgg-verydeep-19.ma  83%[===============>    ] 460.09M  3.07MB/s    eta 24s    


vgg-verydeep-19.mat  83%[===============>    ] 460.76M  3.10MB/s    eta 24s    


gg-verydeep-19.mat   84%[===============>    ] 461.55M  3.18MB/s    eta 24s    


g-verydeep-19.mat    84%[===============>    ] 462.29M  3.23MB/s    eta 23s    


-verydeep-19.mat     84%[===============>    ] 463.01M  3.29MB/s    eta 23s    


verydeep-19.mat      84%[===============>    ] 463.73M  3.26MB/s    eta 23s    


erydeep-19.mat       84%[===============>    ] 464.46M  3.31MB/s    eta 23s    


rydeep-19.mat        84%[===============>    ] 465.10M  3.32MB/s    eta 23s    


ydeep-19.mat         84%[===============>    ] 465.85M  3.34MB/s    eta 22s    


deep-19.mat          84%[===============>    ] 466.56M  3.37MB/s    eta 22s    


eep-19.mat           85%[================>   ] 467.30M  3.44MB/s    eta 22s    


ep-19.mat            85%[================>   ] 467.85M  3.42MB/s    eta 22s    


p-19.mat             85%[================>   ] 468.46M  3.34MB/s    eta 22s    


-19.mat              85%[================>   ] 469.20M  3.39MB/s    eta 21s    


19.mat               85%[================>   ] 469.88M  3.38MB/s    eta 21s    


9.mat                85%[================>   ] 470.54M  3.37MB/s    eta 21s    


.mat                 85%[================>   ] 471.21M  3.35MB/s    eta 21s    


mat                  85%[================>   ] 472.03M  3.41MB/s    eta 21s    


at                   86%[================>   ] 472.48M  3.35MB/s    eta 20s    


t                    86%[================>   ] 473.05M  3.31MB/s    eta 20s    


                     86%[================>   ] 473.49M  3.17MB/s    eta 20s    


                  i  86%[================>   ] 473.87M  3.10MB/s    eta 20s    


                 im  86%[================>   ] 474.27M  3.01MB/s    eta 20s    


                ima  86%[================>   ] 474.63M  2.87MB/s    eta 20s    


               imag  86%[================>   ] 475.09M  2.80MB/s    eta 20s    


              image  86%[================>   ] 475.57M  2.70MB/s    eta 20s    


             imagen  86%[================>   ] 475.99M  2.62MB/s    eta 20s    


            imagene  86%[================>   ] 476.48M  2.58MB/s    eta 20s    


           imagenet  86%[================>   ] 476.85M  2.49MB/s    eta 19s    


          imagenet-  86%[================>   ] 477.24M  2.39MB/s    eta 19s    


         imagenet-v  86%[================>   ] 477.65M  2.29MB/s    eta 19s    


        imagenet-vg  87%[================>   ] 478.09M  2.17MB/s    eta 19s    


       imagenet-vgg  87%[================>   ] 478.38M  2.10MB/s    eta 19s    


      imagenet-vgg-  87%[================>   ] 478.82M  2.06MB/s    eta 19s    


     imagenet-vgg-v  87%[================>   ] 479.35M  2.00MB/s    eta 19s    


    imagenet-vgg-ve  87%[================>   ] 479.84M  1.99MB/s    eta 19s    


   imagenet-vgg-ver  87%[================>   ] 480.30M  1.97MB/s    eta 19s    


  imagenet-vgg-very  87%[================>   ] 480.84M  2.00MB/s    eta 19s    


 imagenet-vgg-veryd  87%[================>   ] 481.32M  2.04MB/s    eta 18s    


imagenet-vgg-veryde  87%[================>   ] 481.79M  2.05MB/s    eta 18s    


magenet-vgg-verydee  87%[================>   ] 482.27M  2.13MB/s    eta 18s    


agenet-vgg-verydeep  87%[================>   ] 482.74M  2.14MB/s    eta 18s    


genet-vgg-verydeep-  87%[================>   ] 483.27M  2.16MB/s    eta 18s    


enet-vgg-verydeep-1  88%[================>   ] 483.77M  2.19MB/s    eta 18s    


net-vgg-verydeep-19  88%[================>   ] 484.29M  2.20MB/s    eta 18s    


et-vgg-verydeep-19.  88%[================>   ] 484.80M  2.23MB/s    eta 18s    


t-vgg-verydeep-19.m  88%[================>   ] 485.35M  2.29MB/s    eta 18s    


-vgg-verydeep-19.ma  88%[================>   ] 485.90M  2.33MB/s    eta 18s    


vgg-verydeep-19.mat  88%[================>   ] 486.43M  2.37MB/s    eta 17s    


gg-verydeep-19.mat   88%[================>   ] 487.05M  2.46MB/s    eta 17s    


g-verydeep-19.mat    88%[================>   ] 487.62M  2.49MB/s    eta 17s    


-verydeep-19.mat     88%[================>   ] 488.18M  2.53MB/s    eta 17s    


verydeep-19.mat      88%[================>   ] 488.74M  2.55MB/s    eta 17s    


erydeep-19.mat       89%[================>   ] 489.29M  2.56MB/s    eta 16s    


rydeep-19.mat        89%[================>   ] 489.91M  2.60MB/s    eta 16s    


ydeep-19.mat         89%[================>   ] 490.29M  2.56MB/s    eta 16s    


deep-19.mat          89%[================>   ] 491.01M  2.61MB/s    eta 16s    


eep-19.mat           89%[================>   ] 491.65M  2.66MB/s    eta 16s    


ep-19.mat            89%[================>   ] 492.24M  2.67MB/s    eta 16s    


p-19.mat             89%[================>   ] 492.87M  2.69MB/s    eta 16s    


-19.mat              89%[================>   ] 493.48M  2.72MB/s    eta 16s    


19.mat               89%[================>   ] 494.10M  2.77MB/s    eta 16s    


9.mat                90%[=================>  ] 494.79M  2.79MB/s    eta 16s    


.mat                 90%[=================>  ] 495.48M  2.86MB/s    eta 15s    


mat                  90%[=================>  ] 496.21M  2.91MB/s    eta 15s    


at                   90%[=================>  ] 496.99M  2.98MB/s    eta 15s    


t                    90%[=================>  ] 497.85M  3.07MB/s    eta 15s    


                     90%[=================>  ] 498.46M  3.06MB/s    eta 15s    


                  i  90%[=================>  ] 499.38M  3.15MB/s    eta 14s    


                 im  91%[=================>  ] 500.43M  3.29MB/s    eta 14s    


                ima  91%[=================>  ] 501.49M  3.42MB/s    eta 14s    


               imag  91%[=================>  ] 502.54M  3.65MB/s    eta 14s    


              image  91%[=================>  ] 503.20M  3.58MB/s    eta 14s    


             imagen  91%[=================>  ] 504.09M  3.68MB/s    eta 12s    


            imagene  91%[=================>  ] 504.74M  3.70MB/s    eta 12s    


           imagenet  92%[=================>  ] 505.70M  3.81MB/s    eta 12s    


          imagenet-  92%[=================>  ] 506.62M  3.89MB/s    eta 12s    


         imagenet-v  92%[=================>  ] 507.60M  3.95MB/s    eta 12s    


        imagenet-vg  92%[=================>  ] 508.55M  4.07MB/s    eta 11s    


       imagenet-vgg  92%[=================>  ] 509.59M  4.18MB/s    eta 11s    


      imagenet-vgg-  92%[=================>  ] 510.59M  4.26MB/s    eta 11s    


     imagenet-vgg-v  93%[=================>  ] 511.60M  4.30MB/s    eta 11s    


    imagenet-vgg-ve  93%[=================>  ] 512.62M  4.37MB/s    eta 11s    


   imagenet-vgg-ver  93%[=================>  ] 513.65M  4.48MB/s    eta 10s    


  imagenet-vgg-very  93%[=================>  ] 514.70M  4.56MB/s    eta 10s    


 imagenet-vgg-veryd  93%[=================>  ] 515.77M  4.60MB/s    eta 10s    


imagenet-vgg-veryde  94%[=================>  ] 516.84M  4.57MB/s    eta 10s    


magenet-vgg-verydee  94%[=================>  ] 517.96M  4.63MB/s    eta 10s    


agenet-vgg-verydeep  94%[=================>  ] 519.10M  4.86MB/s    eta 8s     


genet-vgg-verydeep-  94%[=================>  ] 520.16M  4.90MB/s    eta 8s     


enet-vgg-verydeep-1  94%[=================>  ] 521.29M  5.01MB/s    eta 8s     


net-vgg-verydeep-19  95%[==================> ] 522.38M  5.11MB/s    eta 8s     


et-vgg-verydeep-19.  95%[==================> ] 523.51M  5.17MB/s    eta 8s     


t-vgg-verydeep-19.m  95%[==================> ] 524.66M  5.25MB/s    eta 7s     


-vgg-verydeep-19.ma  95%[==================> ] 525.84M  5.31MB/s    eta 7s     


vgg-verydeep-19.mat  95%[==================> ] 526.98M  5.36MB/s    eta 7s     


gg-verydeep-19.mat   96%[==================> ] 528.10M  5.39MB/s    eta 7s     


g-verydeep-19.mat    96%[==================> ] 529.21M  5.42MB/s    eta 7s     


-verydeep-19.mat     96%[==================> ] 530.32M  5.44MB/s    eta 5s     


verydeep-19.mat      96%[==================> ] 531.40M  5.46MB/s    eta 5s     


erydeep-19.mat       96%[==================> ] 532.49M  5.45MB/s    eta 5s     


rydeep-19.mat        97%[==================> ] 533.62M  5.49MB/s    eta 5s     


ydeep-19.mat         97%[==================> ] 534.51M  5.38MB/s    eta 5s     


deep-19.mat          97%[==================> ] 535.68M  5.36MB/s    eta 4s     


eep-19.mat           97%[==================> ] 536.80M  5.37MB/s    eta 4s     


ep-19.mat            97%[==================> ] 537.94M  5.35MB/s    eta 4s     


p-19.mat             98%[==================> ] 538.95M  5.34MB/s    eta 4s     


-19.mat              98%[==================> ] 539.74M  5.22MB/s    eta 4s     


19.mat               98%[==================> ] 540.60M  5.18MB/s    eta 2s     


9.mat                98%[==================> ] 541.48M  5.11MB/s    eta 2s     


.mat                 98%[==================> ] 542.40M  5.04MB/s    eta 2s     


mat                  98%[==================> ] 543.27M  4.97MB/s    eta 2s     


at                   99%[==================> ] 543.88M  4.76MB/s    eta 2s     


t                    99%[==================> ] 544.45M  4.55MB/s    eta 1s     


                     99%[==================> ] 545.13M  4.40MB/s    eta 1s     


                  i  99%[==================> ] 545.90M  4.33MB/s    eta 1s     


                 im  99%[==================> ] 546.51M  4.16MB/s    eta 1s     


                ima  99%[==================> ] 547.04M  3.97MB/s    eta 1s     


               imag  99%[==================> ] 547.74M  3.93MB/s    eta 0s     


              image  99%[==================> ] 548.55M  3.90MB/s    eta 0s     


             imagen  99%[==================> ] 549.10M  3.63MB/s    eta 0s     


imagenet-vgg-veryde 100%[===================>] 549.36M  3.64MB/s    in 2m 27s  

2021-12-10 15:57:48 (3.73 MB/s) - ‘imagenet-vgg-verydeep-19.mat’ saved [576042600/576042600]



In [ ]:
# Check that we got the .mat file
%ls /notebooks/fast-style-transfer/data

imagenet-vgg-verydeep-19.mat


## 3. Training with style.py

In [ ]:
# We first move into the fast-style-transfer repo
%cd /notebooks/fast-style-transfer

/notebooks/fast-style-transfer


Arguments - make sure 

- the --checkpoint-dir argument points to _/notebooks/fast-style-transfer/ckpt_
- the --style argument points to your style image, such as _/notebooks/fast-stype-transfer/images/style/slovenia-gan.jpg_
- the --vgg-path points to _/notebooks/fast-style-transfer/data/imagenet-vgg-verydeep-19.mat_


You can change the number after --epoch to change the training time, default to 2, it has to be > 0

Keep the below cell running, keep the tab active and wait. It took me 2 hours.

Paperspace might not update the cell output accordingly, so just be patient.

In [ ]:
# Remember to replace the name after 'images/style/' to your own style image name
!python style.py --checkpoint-dir /notebooks/fast-style-transfer/ckpt --style images/style/slovenia-gan.jpg --style-weight 1.5e2 --train-path --vgg-path /notebooks/fast-style-transfer/data/imagenet-vgg-verydeep-19.mat --epochs 2

At this point, you will be able to see 4 files in the '/fast-style-transfer/ckpt' folder.

## 4. Converting the model to ml5js

### Switch back to Tensorflow 1.14.0 

In [ ]:
!pip install tensorflow==1.14.0

### Download the repo for conversion

In [ ]:
# Let's go back to the root folder
%cd /notebooks

# Download the repo
!git clone https://github.com/reiinakano/fast-style-transfer-deeplearnjs.git

# Switch to the repo
%cd fast-style-transfer-deeplearnjs

/notebooks


Cloning into 'fast-style-transfer-deeplearnjs'...


remote: Enumerating objects: 1407, done.
Receiving objects:   0% (1/1407)   
Receiving objects:   1% (15/1407)   
Receiving objects:   2% (29/1407)   
Receiving objects:   3% (43/1407)   


Receiving objects:  16% (239/1407), 21.78 MiB | 21.77 MiB/s   


Receiving objects:  18% (254/1407), 21.78 MiB | 21.77 MiB/s   


Receiving objects:  27% (380/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  28% (394/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  29% (409/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  30% (423/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  31% (437/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  32% (451/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  33% (465/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  34% (479/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  35% (493/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  36% (507/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  37% (521/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  38% (535/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  39% (549/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  40% (563/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  41% (577/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  42% (591/1407), 32.45 MiB | 21.63 M

Receiving objects:  76% (1070/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  77% (1084/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  78% (1098/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  79% (1112/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  80% (1126/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  81% (1140/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  82% (1154/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  83% (1168/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  84% (1182/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  85% (1196/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  86% (1211/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  87% (1225/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  88% (1239/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  89% (1253/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  90% (1267/1407), 32.45 MiB | 21.63 MiB/s   
Receiving objects:  91% (1281/1407), 32.

/notebooks/fast-style-transfer-deeplearnjs


### Convert the model

Make sure that 

- the --output_dir argument points to your _ckpts/model_name_ (this directory will be created by this script)

in both script calls.

In [ ]:
# Change src/ckpts/slo-transfer to src/ckpts/YOUR_OWN_FOLDER
!python scripts/dump_checkpoint_vars.py --output_dir=src/ckpts/slo-transfer --checkpoint_file=/notebooks/fast-style-transfer/ckpt/fns.ckpt

# Change src/ckpts/slo-transfer to src/ckpts/YOUR_OWN_FOLDER
!python scripts/remove_optimizer_variables.py --output_dir=src/ckpts/slo-transfer

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/dtypes.py:516: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/dtypes.py:517: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/dtypes.py:518: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/dtypes.py:519: FutureWarning: Passing (type, 1) or 

/usr/local/lib/python3.6/dist-packages/tensorboard/compat/tensorflow_stub/dtypes.py:541: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
/usr/local/lib/python3.6/dist-packages/tensorboard/compat/tensorflow_stub/dtypes.py:542: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
/usr/local/lib/python3.6/dist-packages/tensorboard/compat/tensorflow_stub/dtypes.py:543: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
/usr/local/lib/python3.6/dist-packages/tensorboard/compat/tensorflow_stub/dtypes.py:544: FutureWa

W1118 21:41:48.192328 139620861376320 deprecation_wrapper.py:119] From scripts/dump_checkpoint_vars.py:48: The name tf.train.NewCheckpointReader is deprecated. Please use tf.compat.v1.train.NewCheckpointReader instead.



W1118 21:41:48.200464 139620861376320 deprecation_wrapper.py:119] From scripts/dump_checkpoint_vars.py:51: The name tf.gfile.MakeDirs is deprecated. Please use tf.io.gfile.makedirs instead.

Writing variable Variable_9/Adam_1...
Writing variable Variable_8/Adam...
Writing variable Variable_8...
Writing variable Variable_7/Adam_1...
Writing variable Variable_7/Adam...
Writing variable Variable_7...
Writing variable Variable_6/Adam...
Writing variable Variable_6...
Writing variable Variable_5/Adam...
Writing variable Variable_47/Adam_1...
Writing variable Variable_45/Adam_1...
Writing variable Variable_45/Adam...
Writing variable Variable_45...
Writing variable Variable_43/Adam_1...
Writing variable Variable_43/Adam...
Writing variable Variable_43...
Writing variable Variable_5/Adam_1...
Writing variable Variable_42/Adam...


Writing variable Variable_41/Adam...
Writing variable Variable_41...
Writing variable Variable_40...
Writing variable Variable_39/Adam_1...
Writing variable Variable_39...
Writing variable Variable_38/Adam_1...
Writing variable Variable_4/Adam_1...
Writing variable Variable_38...
Writing variable Variable_44...
Writing variable Variable_20/Adam...
Writing variable Variable_10/Adam...
Writing variable Variable_37/Adam...
Writing variable Variable_31/Adam...
Writing variable Variable_19/Adam...
Writing variable Variable_25...
Writing variable Variable_9...
Writing variable Variable_32/Adam...
Writing variable Variable_34...


Writing variable Variable_5...
Writing variable Variable_47/Adam...
Writing variable Variable_17/Adam_1...
Writing variable Variable_32...
Writing variable Variable_18/Adam...
Writing variable Variable_17/Adam...
Writing variable Variable_19/Adam_1...
Writing variable Variable_11/Adam_1...
Writing variable Variable_17...
Writing variable Variable_18...
Writing variable Variable_44/Adam_1...
Writing variable Variable_36...
Writing variable Variable_16...
Writing variable Variable_2...


Writing variable Variable_14/Adam_1...
Writing variable beta2_power...
Writing variable Variable_15...
Writing variable Variable_30...
Writing variable Variable_15/Adam_1...
Writing variable Variable_46...
Writing variable Variable_15/Adam...
Writing variable Variable_24/Adam...
Writing variable Variable_16/Adam...
Writing variable Variable_11...
Writing variable Variable_2/Adam_1...
Writing variable Variable_41/Adam_1...
Writing variable Variable_38/Adam...


Writing variable Variable_23...
Writing variable Variable_10/Adam_1...
Writing variable Variable_24...
Writing variable Variable_37/Adam_1...
Writing variable Variable_47...
Writing variable Variable_14...
Writing variable Variable_20...
Writing variable Variable_30/Adam...
Writing variable Variable_1/Adam...
Writing variable Variable_29/Adam_1...
Writing variable Variable/Adam_1...
Writing variable Variable_19...
Writing variable Variable_21/Adam_1...
Writing variable Variable_11/Adam...
Writing variable Variable_29...
Writing variable Variable_22...
Writing variable Variable_16/Adam_1...
Writing variable beta1_power...
Writing variable Variable_40/Adam_1...
Writing variable Variable_1...
Writing variable Variable_44/Adam...
Writing variable Variable_12...


Writing variable Variable_36/Adam...
Writing variable Variable/Adam...
Writing variable Variable_25/Adam...
Writing variable Variable_12/Adam_1...
Writing variable Variable_23/Adam_1...
Writing variable Variable_12/Adam...
Writing variable Variable_21...
Writing variable Variable_27...
Writing variable Variable_4...
Writing variable Variable_37...
Writing variable Variable_10...
Writing variable Variable_13...
Writing variable Variable_28/Adam...
Writing variable Variable_33/Adam_1...
Writing variable Variable_13/Adam_1...


Writing variable Variable_14/Adam...
Writing variable Variable_3/Adam_1...
Writing variable Variable_18/Adam_1...
Writing variable Variable_2/Adam...
Writing variable Variable_31/Adam_1...
Writing variable Variable_9/Adam...
Writing variable Variable_42/Adam_1...
Writing variable Variable_22/Adam...
Writing variable Variable_6/Adam_1...
Writing variable Variable_22/Adam_1...
Writing variable Variable_23/Adam...
Writing variable Variable_28/Adam_1...
Writing variable Variable_35/Adam_1...
Writing variable Variable_20/Adam_1...
Writing variable Variable_26/Adam...
Writing variable Variable_40/Adam...
Writing variable Variable_4/Adam...
Writing variable Variable_30/Adam_1...
Writing variable Variable_33/Adam...
Writing variable Variable_26/Adam_1...
Writing variable Variable_26...
Writing variable Variable_27/Adam_1...
Writing variable Variable_1/Adam_1...
Writing variable Variable_29/Adam...
Writing variable Variable_25/Adam_1...
Writing variable Variable_28...
Writing variable Variable_

Writing variable Variable_13/Adam...
Writing variable Variable_27/Adam...
Writing variable Variable_33...
Writing variable Variable_34/Adam...
Writing variable Variable_34/Adam_1...
Writing variable Variable_46/Adam_1...
Writing variable Variable_24/Adam_1...
Writing variable Variable_35...
Writing variable Variable_35/Adam...
Writing variable Variable_46/Adam...
Writing variable Variable...
Writing variable Variable_36/Adam_1...
Writing manifest to src/ckpts/slo-gan/manifest.json


Done!


#### Zip the model and download it

Again, make sure the paths are correct for your model.

`tar cfz /notebooks/fast-style-transfer-deeplearnjs/src/ckpts/YOUR-MODEL-FOLDER.tar.gz /notebooks/fast-style-transfer-deeplearnjs/src/ckpts/YOUR-MODEL-FOLDER`

In [ ]:
# We use tar because no zip support
# Change /slo-transfer.tar.gz to /YOUR_FOLDER_NAME.tar.gz, same as src/ckpts/slo-transfer to /YOUR_FOLDER_NAME
!tar czf /notebooks/fast-style-transfer-deeplearnjs/src/ckpts/slo-transfer.tar.gz /notebooks/fast-style-transfer-deeplearnjs/src/ckpts/slo-transfer

tar: Removing leading `/' from member names


Now you should be able to see a YOUR_MODEL_NAME.tar.gz archived file from the Files panel (you might need to refresh the whole page if you don't see it there). Download it from the panel.

## 5. Run the model in ml5js

Put the downloaded and unzipped model folder into your [p5 sketch](https://editor.p5js.org/ml5/sketches/StyleTransfer_Image) in the _models/_ directory.

Change the model path in your p5 sketch:  
`style = ml5.styleTransfer('models/YOUR_MODEL_NAME', modelLoaded);`

## 6. Using the model for video style transfer

Make sure you uploaded the video you'd like to transfer the style to. You can upload it to the root folder (i.e. inside /notebooks)

Make sure you installed the required libraries before. You can also do so right here.

_Note: You do not need to convert the model to ml5 to use it for video style transfer here._

In [ ]:
# Install libraries for video style transfer
!pip install ffmpeg
!pip install moviepy==1.0.2
!pip install scipy
# Downgrading the image-ffmpeg prevent some pipe error
!pip install imageio-ffmpeg==0.2.0

We also need to install TensorFlow 2.10. version again. You can skip this step if you did not convert the model to ml5.

In [ ]:
!pip install tensorflow-gpu==2.1.0

#### Create the output-video directory

In [ ]:
%mkdir /notebooks/output-video

#### Style transfer!

Again - paths!

- the --in-path should point to your video, e.g. _/notebooks/MY_ORIGINAL_VIDEO.mp4_
- the --out-path should point to the _/notebooks/output-video/MY_TRANSFERED_VIDEO.mp4_

In [ ]:
# Go into the fast-style-transfer dir
%cd /notebooks/fast-style-transfer/

In [ ]:
# From here we run out process
# as stated here https://github.com/lengstrom/fast-style-transfer#stylizing-video

!python transform_video.py --in-path /notebooks/maribor.mp4 \
  --checkpoint ckpt/ \
  --tmp-dir /notebooks/tmp-out/ \
  --out-path /notebooks/output-video/video.mp4 \
  --device /gpu:0 \
  --batch-size 4

You should now see your style-transfered video in the output-video directory. Download it and enjoy!

***

Made by @zalashub on [GitHub](https://github.com/zalashub) and [Twitter](https://twitter.com/zalashub).